ra/ra.go

```diff
@@ -1960,18 +1960,25 @@ func (ra *RegistrationAuthorityImpl) DeactivateRegistration(ctx context.Context,
 }
 
 // DeactivateAuthorization deactivates a currently valid authorization
-func (ra *RegistrationAuthorityImpl) DeactivateAuthorization(ctx context.Context, auth core.Authorization) error {
-	if auth.Status != core.StatusValid && auth.Status != core.StatusPending {
-		return berrors.MalformedError("only valid and pending authorizations can be deactivated")
+func (ra *RegistrationAuthorityImpl) DeactivateAuthorization(ctx context.Context, req *corepb.Authorization) (*emptypb.Empty, error) {
+	if req == nil || req.Id == "" || req.Identifier == "" || req.Status == "" || req.Expires == 0 {
+		return nil, errIncompleteGRPCRequest
 	}
-	authzID, err := strconv.ParseInt(auth.ID, 10, 64)
+	authz, err := bgrpc.PBToAuthz(req)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	if authz.Status != core.StatusValid && authz.Status != core.StatusPending {
+		return nil, berrors.MalformedError("only valid and pending authorizations can be deactivated")
+	}
+	authzID, err := strconv.ParseInt(authz.ID, 10, 64)
```

Comment:
Can skip this entirely, since we'll just be passing the `req.Id` through into the `sapb` protobuf verbatim.

Reply:
Got it, keeping the `int64` parsing though.